crates/solidity/outputs/cargo/tests/src/doc_examples/query_language.rs

```diff
@@ -0,0 +1,286 @@
+use once_cell::sync::Lazy;
+use regex::Regex;
+use semver::Version;
+use slang_solidity::kinds::RuleKind;
+use slang_solidity::language::Language;
+use slang_solidity::query::{Query, QueryResultIterator};
```

Comment:
suggestion: the modules under `src/doc_examples` have 1:1 relationship with the user guides markdown, and the typescript examples. This makes it easy to edit/refactor them.
If the Rust query-related examples are growing into multiple modules, what do you think about splitting it into multiple sub-modules under `src/doc_examples/using_queries/mod.rs`?